### PR TITLE
optimize: optimize `httpcore` and `httpclient` dependencies

### PR DESCRIPTION
**Pull request by wangliang181230**
optimize: fix the definition of `httpcore` and `httpclient` dependencies.
修正 `httpcore` 和 `httpclient` 依赖项的定义方式，避免基于springboot3.1.x的单测失败。

`develop` 分支改了的，`2.x` 分支里给忘掉了，所以补了这个PR。

# 错误日志：
https://github.com/wangliang181230/seata/actions/runs/6272799601/job/17084849769

# actions测试成功记录：
all：https://github.com/wangliang181230/seata/actions
`test.yml`：https://github.com/wangliang181230/seata/actions/runs/6293863985
`test-druid.yml`：https://github.com/wangliang181230/seata/actions/runs/6293863984

### CODE REVIEW

File: dependencies/pom.xml [hunk 1]
```diff
@@ -35,6 +35,7 @@
         <dubbo-seata.version>1.0.2</dubbo-seata.version>
         <brpc.version>2.5.9</brpc.version>
         <hsf.version>1.8.3</hsf.version>
+        <bytebuddy.version>1.12.17</bytebuddy.version>
```
Comment (wangliang181230) [inline dependencies/pom.xml:38]: 从下面移上来。和develop分支位置保持一致。方便比对。

**Review by a364176773 — APPROVED** [review]
LGTM

**Review by slievrly — APPROVED** [review]
LGTM

File: dependencies/pom.xml [hunk 2]
```diff
@@ -81,6 +82,10 @@
         <zstd.version>1.5.0-4</zstd.version>
         <xstream.version>1.4.20</xstream.version>
 
+        <!-- The `httpcore` and `httpclient` have been removed in spring-boot 3.1.0 and above. -->
+        <httpcore.version>4.4.16</httpcore.version>
```
Comment (slievrly) [inline dependencies/pom.xml:86]: Is it the same as the implicit dependency version?

Comment (wangliang181230) [reply dependencies/pom.xml:86]: Different versions

Comment (wangliang181230) [reply dependencies/pom.xml:85]: It is the same version as in the later version of Spring Boot.
I tested the lower version of Spring Boot and found no issues.

**Pull request merged**